mavros_extras/src/plugins/esc_telemetry.cpp

```diff
@@ -21,6 +25,60 @@ namespace mavros
 {
 namespace extra_plugins
 {
+// for all the dignostics parameters
+struct Limits {
+	bool _diag_enabled;
+	// Temperature checks
+	float _temp_min_nok;
+	float _temp_min_ok;
+	float _temp_max_nok;
+	float _temp_max_ok;
+	// Voltage checks
+	float _volt_min_nok;
+	float _volt_min_ok;
+	float _volt_max_nok;
+	float _volt_max_ok;
+	// Current checks
+	float _curr_min_nok;
+	float _curr_min_ok;
+	float _curr_max_nok;
+	float _curr_max_ok;
+	// RPM checks
+	int _rpm_min_nok;
+	int _rpm_min_ok;
+	int _rpm_max_nok;
+	int _rpm_max_ok;
+	// Count checks, should be uint16_t but ROS only allows int parameters
+	int _count_min_delta;
+
+	explicit Limits(ros::NodeHandle &pnh) {
+		if (!pnh.getParam("enabled", this->_diag_enabled)) {
+			ROS_FATAL_STREAM("Could not retrieve enabled parameter from parameter server");
+			// TODO: how to gracefully exit here???
+		}
+		if (this->_diag_enabled)
+		{
+			pnh.param("temp_min/nok", this->_temp_min_nok, 0.0f);
+			pnh.param("temp_min/ok", this->_temp_min_ok, 1.0f);
+			pnh.param("volt_min/nok", this->_volt_min_nok, 14.0f);
+			pnh.param("volt_min/ok", this->_volt_min_ok, 15.0f);
+			pnh.param("curr_min/nok", this->_curr_min_nok, 4.0f);
+			pnh.param("curr_min/ok", this->_curr_min_ok, 5.0f);
+			pnh.param("rpm_min/nok", this->_rpm_min_nok, 2000);
+			pnh.param("rpm_min/ok", this->_rpm_min_ok, 3000);
+			pnh.param("temp_max/nok", this->_temp_max_nok, 90.0f);
+			pnh.param("temp_max/ok", this->_temp_max_ok, 85.0f);
+			pnh.param("volt_max/nok", this->_volt_max_nok, 17.0f);
+			pnh.param("volt_max/ok", this->_volt_max_ok, 16.0f);
+			pnh.param("curr_max/nok", this->_curr_max_nok, 10.0f);
+			pnh.param("curr_max/ok", this->_curr_max_ok, 8.0f);
+			pnh.param("rpm_max/nok", this->_rpm_max_nok, 12000);
+			pnh.param("rpm_max/ok", this->_rpm_max_ok, 9000);
```

Comment:
Perhaps i'd generate that list by cog.
```
// [[[cog:
// fields = [
//   ("temp", (1.0, 0.0), (85.0, 90.0), ),
// ...
// ]
//
// for field, min_, max_ = range fields:
//    for fm, lim = zip(("min", "max"), (min_, max_)):
//        for fn, l = zip(("ok", "nok"), lim):
//            cog.outl(f"""pnh.param("{field}_{fm}/{fn}", this->{field}_{fm}_{fn}, {l});""")
// ]]]
```

Reply:
done, now I just need to find out how to run the cog compiler

Reply:
`cog -cr <file>`

https://github.com/mavlink/mavros/blob/ros2/cog-requirements.txt

Reply:
Fixed it, but now it has other issues :(